### PR TITLE
Clickable InlayHint for Java type

### LINKED ISSUE #702: `variable??` cannot be resolved or is not a field of `ClassX` Java type.qute(UnknownProperty)
Example:
{#formselect value=company.address.country.id?? /}

`id??` will get marked in red:
`id??` cannot be resolved or is not a field of `model.Country` Java type.qute(UnknownProperty)

### PR DESCRIPTION
**Pull request by angelozerr**
Clickable InlayHint for Java type

Fixes #702

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by datho7561** [comment]
Is this PR ready? If so, could you please mark it as ready for review?

**Comment by angelozerr** [comment]
Yes it is ready. I switched to real pr

File: src/qute/languageServer/inlayHintsProvider.ts [hunk 1]
```diff
@@ -54,42 +54,88 @@ export class QuteInlayHintsProvider implements InlayHintsProvider {
...
+    : item.label.map(asInlayHintLabelPart);
+  const result = new InlayHint(client.protocol2CodeConverter.asPosition(item.position), label);
+  result.paddingRight = true;
+  result.kind = InlayHintKind.Parameter;
+  return result;
+}
+
+function asInlayHintLabelPart(item: LSInlayHintLabelPart): InlayHintLabelPart {
+  const result = new InlayHintLabelPart(item.value);
+  //if (item.location !== undefined) { result.location = asLocation(item.location); }
```
Comment (datho7561) [inline src/qute/languageServer/inlayHintsProvider.ts]: remove this comment

File: src/qute/languageServer/inlayHintsProvider.ts [hunk 2]
```diff
@@ -54,42 +54,88 @@ export class QuteInlayHintsProvider implements InlayHintsProvider {
...
+
+export type LSInlayHintLabelPart = {
+
+  /**
+   * The value of this label part.
+   */
+  value: string;
+
+  /**
+   * The tooltip text when you hover over this label part. Depending on
```
Comment (datho7561) [inline src/qute/languageServer/inlayHintsProvider.ts]: Fix this comment

**Review by datho7561 — CHANGES_REQUESTED** [review]
There are a few comments that need to be fixed, but otherwise it appears to be good

File: src/qute/languageServer/inlayHintsProvider.ts (hunk 1 above)
Comment (angelozerr) [reply src/qute/languageServer/inlayHintsProvider.ts]: fixed

File: src/qute/languageServer/inlayHintsProvider.ts (hunk 2 above)
Comment (angelozerr) [reply src/qute/languageServer/inlayHintsProvider.ts]: fixed

**Review by datho7561 — APPROVED** [review]
Looks good to me! Thanks, Angelo!

**Comment by datho7561** [comment]
Requires redhat-developer/quarkus-ls#704

**Pull request merged**